spring-observability-core/src/main/java/org/springframework/observability/tracing/listener/HttpTracingRecordingListener.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2021-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.observability.tracing.listener;
+
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+
+import org.springframework.observability.core.http.HttpRequest;
+import org.springframework.observability.core.http.HttpResponse;
+import org.springframework.observability.event.interval.IntervalEvent;
+import org.springframework.observability.event.interval.IntervalRecording;
+import org.springframework.observability.lang.Nullable;
+import org.springframework.observability.tracing.CurrentTraceContext;
+import org.springframework.observability.tracing.Span;
+
+abstract class HttpTracingRecordingListener<CONTEXT, REQ extends HttpRequest, RES extends HttpResponse> {
+
+	private final CurrentTraceContext currentTraceContext;
+
+	private final Function<REQ, Span> startFunction;
+
+	private final BiConsumer<RES, Span> stopConsumer;
+
+	private final TracingTagFilter tracingTagFilter = new TracingTagFilter();
+
+	HttpTracingRecordingListener(CurrentTraceContext currentTraceContext, Function<REQ, Span> startFunction,
+			BiConsumer<RES, Span> stopConsumer) {
+		this.currentTraceContext = currentTraceContext;
+		this.startFunction = startFunction;
+		this.stopConsumer = stopConsumer;
+	}
+
+	void doOnStart(IntervalRecording<CONTEXT> intervalRecording) {
```

Comment:
What do you think about making this a real listener (`implements ...`) so that the classes that extend this does not need to call `doOnStart`/`doOnStop` they just simply don't override the methods or they can call `super.onStart`/`super.onStop` if needed.